server/src/main/java/org/elasticsearch/action/admin/cluster/stats/TransportClusterStatsAction.java

```diff
@@ -136,10 +170,16 @@ protected void newResponseAsync(
             clusterService.threadPool().absoluteTimeInMillis()
         );
 
+        final Map<String, ClusterStatsResponse.RemoteClusterStats> remoteClusterStats =
+            (listener instanceof ActionListenerWithRemotes listenerWithRemotes)
```

Comment:
Hm this type check seems unnecessary, let's find a way to return `Map.of()` on the remotes-disabled path instead.

Reply:
The reason for type check here is because the signature of `newResponseAsync` is fixed so I can't have the correct type there. But I am not sure I like the way it's done now, I will take another look into it.